src/prefect/filesystems.py

```diff
@@ -138,18 +136,14 @@ async def get_directory(
             # and we avoid shutil.copytree raising an error
             return
 
-        if sys.version_info < (3, 8):
-            shutil.copytree(from_path, local_path)
-        else:
-            shutil.copytree(from_path, local_path, dirs_exist_ok=True)
+        copytree(from_path, local_path, dirs_exist_ok=True)
 
-    async def _get_ignore_func(self, local_path: str, ignore_file: str):
+    async def _get_ignore_func(self, ignore_file: str):
         with open(ignore_file, "r") as f:
             ignore_patterns = f.readlines()
 
-        included_files = filter_files(local_path, ignore_patterns)
-
         def ignore_func(directory, files):
+            included_files = filter_files(directory, ignore_patterns)
```

Comment:
I wondered if a better solution to this bit would be to keep the call to `filter_files` in `_get_ignore_func` and then modify `ignore_func` as below:
```
def ignore_func(directory, files):
    relative_path = Path(directory).relative_to(local_path)
    return_val = [f for f in files if str(relative_path / f) not in included_files]
    return return_val
```

The list of included files would be generated once, relative to the root directory where the `.prefectignore` file lives, and then `ignore_func` checks each file against that list of relative paths (rather than just checking the filename with no path like it does in the original code).

This solution allows you to write patterns in your `.prefectignore` that take advantage of the directory structure beyond the current directory that `ignore_func` is running for, e.g.
```
/some/dir/relative/to/ignore/file
**/subdir/*
```

Reply:
I like this! Just made the update and added a test.